### PR TITLE
Refactor: remove all legacy build logic from codebase

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

- Remove all legacy build flags and dependencies from the codebase
- Remove now unused internal maps: `chunkToReferenceIdMap`, `astroStyleMap`, `astroPageStyleMap`
- Remove `vite-plugin-build-html` and associated npm dependencies: `srcset-parse`, `parse5`, `@web/parse5-utils`
- Remove unused `astro-style` prop on generated style tags

## Testing

- Small update to `0-css.test.js` to support removed `astro-style` prop

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/runtime/server/index.ts [hunk 1]
```diff
@@ -546,10 +546,8 @@ export async function renderHead(result: SSRResult): Promise<string> {
 	const styles = Array.from(result.styles)
 		.filter(uniqueElements)
 		.map((style) => {
-			// TODO: clean up legacyBuild from metadata
-			const styleChildren = !result._metadata.legacyBuild ? '' : style.children;
 			return renderElement('style', {
```
Comment (JuanM04) [inline packages/astro/src/runtime/server/index.ts:549]: Is this hole .map() still needed?

**Review by JuanM04 — COMMENTED** [review]
Seems good. Left some comments

File: packages/astro/src/runtime/server/index.ts (hunk 1 above)
Comment (bholmesdev) [reply packages/astro/src/runtime/server/index.ts:549]: I was wondering that too. It looks like this converts each `style` (which is an `SSRElement` object) to an HTML element with `style.props` applied. I'd assume it's still necessary for that

Comment (matthewp) [reply packages/astro/src/runtime/server/index.ts:549]: It's not needed

**Comment by matthewp** [comment]
vite-plugin-build-html can be removed entirely (and any dependencies that only it uses), and the logic in core/build/index.ts can be updated so that path is not taken.

**Review by matthewp — CHANGES_REQUESTED** [review]
Remove vite-plugin-build-html and associated logic.

**Comment by FredKSchott** [comment]
Screen Shot 2022-04-19 at 8 48 55 AM

This is very exciting :)

**Comment by bholmesdev** [comment]
@FredKSchott Removed 3 dependencies from Astro core too 🔥

File: packages/astro/src/runtime/server/index.ts (hunk 1 above)
Comment (bholmesdev) [reply packages/astro/src/runtime/server/index.ts]: Removed!

File: packages/astro/package.json [hunk 2]
```diff
@@ -109,7 +108,6 @@
     "micromorph": "^0.1.2",
     "mime": "^3.0.0",
     "ora": "^6.1.0",
-    "parse5": "^6.0.1",
```
Comment (matthewp) [inline packages/astro/package.json:112]: glad to not need to do this any more

**Review by matthewp — APPROVED** [review]

**Pull request merged**